### PR TITLE
Call pytest_report_collectionfinish hook when --collect-only is passed

### LINKED ISSUE #2895: pytest_report_collectionfinish does not execute with --collect-only
As commented [here](https://github.com/pytest-dev/pytest/pull/2623#issuecomment-341478287).

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #2895

### CODE REVIEW

File: testing/test_terminal.py [hunk 1]
```diff
@@ -649,7 +649,8 @@ def test_more_quiet_reporting(self, testdir):
         assert "===" not in s
         assert "passed" not in s
 
-    def test_report_collectionfinish_hook(self, testdir):
+    @pytest.mark.parametrize("param", ["", "--collect-only"])
```
Comment (asottile) [inline testing/test_terminal.py]: I think you can `"param", [(), ("--collect-only",)] ` and then you don't need the ternary below

Comment (nicoddemus) [reply testing/test_terminal.py]: I did this initially, but then the parametrized node id becomes "param0" and "param1", that's why I took this approach. Still want me to change it?

Comment (RonnyPfannschmidt) [reply testing/test_terminal.py]: @nicoddemus you can still pass ids in using ids or param(..., id)

Comment (nicoddemus) [reply testing/test_terminal.py]: duh of course, will change.

Comment (nicoddemus) [reply testing/test_terminal.py]: Fixed, thanks guys. 👍

**Comment by nicoddemus** [comment]
> Just for my curiosity, what's the difference between option.collectonly and getoption("collect-only")?

None, except the former is not really recommended anymore, not sure what's the specific reason for it.

**Pull request merged**